### ISSUE TITLE
Stability issues

### ISSUE THREAD

**Issue by shizzard** [shizzard/marvin#31]
Review and rethink supervision tree to ensure everything will eventually get up in case of errors.

**Comment by shizzard** [comment]
```
marvin_1      | 13:40:32.466 [error] [ISSUE-31] Shard 'marvin_shard_session_00' got message create with undefined channel faked to 0; Struct: #{}
marvin_1      | 15:07:12.599 [error] [ISSUE-31] Shard 'marvin_shard_session_00' got message create with undefined channel faked to 0; Struct: #{}
marvin_1      | 15:55:43.449 [error] [ISSUE-31] Shard 'marvin_shard_session_00' got message create with undefined channel faked to 0; Struct: #{}
```

**Comment by shizzard** [comment]
Erroneous `message_create` PDU:
```json
{"type":0,"tts":false,"timestamp":"2018-05-30T15:55:43.379000+00:00","pinned":false,"nonce":"451413407787122688","mentions":[],"mention_roles":[],"mention_everyone":false,"member":{"roles":["399557799811612704","427217463693017088","399587748283154432"],"nick":null,"mute":false,"joined_at":"2017-05-28T12:32:43.223000+00:00","deaf":false},"id":"451413408236044288","guild_id":"318325967602843658","embeds":[{"url":"https://www.dropbox.com/s/uqla3j21iv2as56/Screenshot%202018-05-30%2018.55.08.png?dl=0","type":"article","title":"Screenshot 2018-05-30 18.55.08.png","thumbnail":{"width":1024,"url":"https://www.dropbox.com/temp_thumb_from_token/s/uqla3j21iv2as56?preserve_transparency=False&size=1024x1024&size_mode=2","proxy_url":"https://images-ext-1.discordapp.net/external/Eqyt_ttmojkOTA6oI13aSCrp5xXmJtHPddqBbOv1Asg/%3Fpreserve_transparency%3DFalse%26size%3D1024x1024%26size_mode%3D2/https/www.dropbox.com/temp_thumb_from_token/s/uqla3j21iv2as56","height":1024},"provider":{"url":null,"name":"Dropbox"},"description":"Shared with Dropbox"}],"edited_timestamp":null,"content":"https://www.dropbox.com/s/uqla3j21iv2as56/Screenshot%202018-05-30%2018.55.08.png?dl=0","channel_id":"448788687300591618","author":{"username":"BizkitCake","id":"143022070429646848","discriminator":"8836","avatar":"d526eca4a94c1810d72f9b0aebb12f4d"},"attachments":[]}
```

**Comment by shizzard** [comment]
`presence_update`:

```json
{
	"user": {
		"id": "321644406979428354"
	},
	"status": "offline",
	"roles": [],
	"guild_id": "318325967602843658",
	"game": null
}
```

```json
{
	"user": {
		"id": "321283856017784833"
	},
	"status": "online",
	"roles": [],
	"nick": null,
	"guild_id": "318325967602843658",
	"game": {
		"type": 3,
		"name": "на битовые часы: 22:50 UTC"
	}
}
```

**Comment by shizzard** [comment]
`message_create`:
```json
{
    "op": 0,
    "d": {
        "type": 0,
        "tts": false,
        "timestamp": "2018-06-01T18:07:46.888000+00:00",
        "pinned": false,
        "nonce": null,
        "mentions": [],
        "mention_roles": [],
        "mention_everyone": false,
        "member": {
            "roles": [
                "451421298204213258",
                "448779403925651463"
            ],
            "nick": null,
            "mute": false,
            "joined_at": "2018-05-23T09:29:07.808830+00:00",
            "deaf": false
        },
        "id": "452171417572868107",
        "guild_id": "318325967602843658",
        "embeds": [
            {
                "url": "https://www.twitch.tv/pur_purr_purrr",
                "type": "rich",
                "title": "Пуррхай как бабочка, жаль что не дайм | ОВЕРВОТЧ!",
                "thumbnail": {
                    "width": 0,
                    "url": "https://static-cdn.jtvnw.net/jtv_user_pictures/9a5c1439-34c4-489d-9af3-0bd854ee574f-profile_image-300x300.jpg",
                    "proxy_url": "https://images-ext-1.discordapp.net/external/b27_zcOQnh-M6AAIk5QqxKacpMLDIRJ-L7hfKsy9JIk/https/static-cdn.jtvnw.net/jtv_user_pictures/9a5c1439-34c4-489d-9af3-0bd854ee574f-profile_image-300x300.jpg",
                    "height": 0
                },
                "image": {
                    "width": 0,
                    "url": "https://static-cdn.jtvnw.net/previews-ttv/live_user_pur_purr_purrr-320x180.jpg?r=217629",
                    "proxy_url": "https://images-ext-2.discordapp.net/external/QRefLHsRB68djPrlssICGc97mZx4uqlvYPfYSgIMDBI/%3Fr%3D217629/https/static-cdn.jtvnw.net/previews-ttv/live_user_pur_purr_purrr-320x180.jpg",
                    "height": 0
                },
                "fields": [
                    {
                        "value": "Overwatch",
                        "name": "Game",
                        "inline": true
                    },
                    {
                        "value": "1",
                        "name": "Viewers",
                        "inline": true
                    }
                ],
                "color": 6570404,
                "author": {
                    "proxy_icon_url": "https://images-ext-1.discordapp.net/external/b27_zcOQnh-M6AAIk5QqxKacpMLDIRJ-L7hfKsy9JIk/https/static-cdn.jtvnw.net/jtv_user_pictures/9a5c1439-34c4-489d-9af3-0bd854ee574f-profile_image-300x300.jpg",
                    "name": "pur_purr_purrr",
                    "icon_url": "https://static-cdn.jtvnw.net/jtv_user_pictures/9a5c1439-34c4-489d-9af3-0bd854ee574f-profile_image-300x300.jpg"
                }
            }
        ],
        "edited_timestamp": null,
        "content": "Клац-клац, хлопци и дивчины, pur_purr_purrr запустила стримулю https://www.twitch.tv/pur_purr_purrr ! Бронза фтв!",
        "channel_id": "399552308486340608",
        "author": {
            "username": "MEE6",
            "id": "159985870458322944",
            "discriminator": "4876",
            "bot": true,
            "avatar": "8c7187aa71f15ba271c85bfece9b067e"
        },
        "attachments": []
    },
    "s": 1,
    "t": "MESSAGE_CREATE"
}

```

**Comment by shizzard** [comment]
Will be fixed within #57.
